### PR TITLE
Fix Silent Authentication Failure in Safari v13.1, Mobile Safari, Brave

### LINKED ISSUE #29: Please log-in again
**Describe the bug**
Every time I refresh or return to the page it asks me to "Please log-in again"

**Expected behavior**
I have no issues with other sites as they let me return to the page without needing to login again. 

**Desktop (please complete the following information):**
 - Browser [Brave]
 - Version [Version 0.68.132 Chromium: 76.0.3809.132 (Official Build) (64-bit)]

**Additional Info**
I am using the Brave browser, which is built off of Chromium/Chrome. Note: I have "Block cross-site cookies" enabled, which I suspect could be a potential place for issue with GitHub's authentication service. Would be happy to help troubleshoot.

### PR DESCRIPTION
**Pull request by NotMyself**
OK @crtr0, here is what I found.

On March 24th, WebKit [released](https://webkit.org/blog/10218/full-third-party-cookie-blocking-and-more/) a change to how samesite cookies are handled for privacy reasons. This causes the latest versions of Safari, mobile Safari and looks like Brave to no longer send cookies that are required by our silent authentication method.

Auth0 SPA SDK, was updated in version 1.8.1 to use a Refresh Token based work around. Which you discovered in that blog post. I honestly had not encountered the issue yet as I typically use Firefox.

What does this mean? The SDK when configured appropriately, in addition to requesting an access will now request a refresh token as well. This RT can be used to request a new access tokens. Which sidesteps the samesite blocking issue in these browsers.

Now, you have a refresh token. But when you refresh the page, all your tokens are lost if using the default token storage mechanism in the SDK. So you never get a chance to use it.

The first change needed it to switch to local storage.

```javascript
const AUTH_CONFIG = {
   ...
    useRefreshTokens: process.env.REACT_APP_USE_REFRESH_TOKENS,
    cacheLocation: process.env.REACT_APP_CACHE_LOCATION,
};

export default AUTH_CONFIG;
```

This comes with [some caveats](https://auth0.com/docs/tokens/concepts/token-storage#browser-local-storage-scenarios) you need to be aware of. Traditionally, delivering a refresh token to a SPA was frowned upon because the blast radius of a RT is pretty huge. A RT is basically a "make as many access tokens as I want" magic wand. So if it leaks, you are in big trouble.

That is where [Refresh Token Rotation](https://auth0.com/docs/tokens/concepts/refresh-token-rotation) comes in. [Enabling RTR](https://auth0.com/docs/tokens/guides/configure-refresh-token-rotation), causes a new refresh token to be issued every time a client exchanges a refresh token to get a new access token.  The previous RT is then invalid.

Which means leaking refresh tokens are no dangerous than leaking access tokens. The blast radius is greatly reduced.

The other change, I made was to externalize your configuration to a .env file based on [this guidance](https://create-react-app.dev/docs/adding-custom-environment-variables/) in the create react app docs.

You will need to run a `cp .env-sample .env` in the root of your project the first time you pull it down. I went ahead and put your values in there, but you might want to get rid of your client specific details. Up to you.

Hope that helps.

Fixes #29

### CODE REVIEW

**Comment by crtr0** [comment]
@NotMyself Thanks for looking into this! I have a question: do you know if there is a recommended way to only enable localstorage for the subset of clients (browsers) that need it? I have read that storing these token in localstorage can be a security risk, so it would be nice to only do it for a subset of clients that require it.

File: src/index.tsx [hunk 1]
```diff
@@ -49,6 +49,7 @@ ReactDOM.render(
         client_id={AUTH_CONFIG.clientId}
         audience={AUTH_CONFIG.audience}
         redirect_uri={`${window.location.origin}/home`}
+        cacheLocation={AUTH_CONFIG.cacheLocation}
```
Comment (NotMyself) [inline src/index.tsx:52]: You could try doing a webkit check here.

Comment (crtr0) [reply src/index.tsx:52]: Ok, thank you!

**Comment by crtr0** [comment]
@NotMyself, I'd like to merge this PR. Can you tweak a few things?

- remove package.json (we use yarn)
- remove the `.env` file and place those values in `.env.local`
- replace `REACT_APP` with `AUTH0` in the config file

It's much appreciated!

**Comment by NotMyself** [comment]
REACT_APP is required. But other than that sure.

**Comment by crtr0** [comment]
> REACT_APP is [required](https://create-react-app.dev/docs/adding-custom-environment-variables/#adding-development-environment-variables-in-env). But other than that sure.

REACT_APP is not required, because I'm not using Create React App. You'll see in the existing `.env.local` that I have a variable called `API_URL`, which is processed just fine when the React app is compiled by Parcel.js.

**Comment by crtr0** [comment]
@NotMyself I just checked-out your branch and can make the additional changes. Thanks a bunch!

**Comment by NotMyself** [comment]
Sounds good. Sorry for the confusion.

**Pull request merged**